opentelemetry-sdk/src/opentelemetry/sdk/metrics/export/aggregate.py

```diff
@@ -57,185 +77,146 @@ def __init__(self, config=None):
 
     def merge(self, other):
-        if verify_type(self, other):
+        if self._verify_type(other):
             with self._lock:
-                self.checkpoint = self._merge_checkpoint(
-                    self.checkpoint, other.checkpoint
-                )
-                self.last_update_timestamp = get_latest_timestamp(
-                    self.last_update_timestamp, other.last_update_timestamp
+                self.checkpoint = self._TYPE(
+                    min(self.checkpoint.min, other.checkpoint.min),
+                    max(self.checkpoint.max, other.checkpoint.max),
+                    self.checkpoint.sum + other.checkpoint.sum,
+                    self.checkpoint.count + other.checkpoint.count,
                 )
+                super().merge(other)
 
 
 class HistogramAggregator(Aggregator):
     """Agregator for ValueRecorder metrics that keeps a histogram of values."""
 
     def __init__(self, config=None):
         super().__init__(config=config)
-        self._lock = threading.Lock()
-        self.last_update_timestamp = None
-        boundaries = self.config.get("bounds")
-        if boundaries and self._validate_boundaries(boundaries):
-            self._boundaries = boundaries
-        else:
-            # no buckets except < 0 and >
-            self._boundaries = (0,)
-
-        self.current = OrderedDict([(bb, 0) for bb in self._boundaries])
-        self.checkpoint = OrderedDict([(bb, 0) for bb in self._boundaries])
-
-        self.current[">"] = 0
-        self.checkpoint[">"] = 0
-
-    # pylint: disable=R0201
-    def _validate_boundaries(self, boundaries):
-        if not boundaries:
-            logger.warning("Bounds is empty. Using default.")
-            return False
-        if not all(
-            boundaries[ii] < boundaries[ii + 1]
-            for ii in range(len(boundaries) - 1)
-        ):
-            logger.warning(
-                "Bounds must be sorted in increasing order. Using default."
-            )
-            return False
-        return True
-
-    @classmethod
-    def _merge_checkpoint(cls, val1, val2):
-        if val1.keys() == val2.keys():
-            for ii, bb in val2.items():
-                val1[ii] += bb
-        else:
-            logger.warning("Cannot merge histograms with different buckets.")
-        return val1
+        # no buckets except < 0 and >
+        bounds = (0,)
+        config_bounds = self.config.get("bounds")
+        if config_bounds is not None:
+            if all(
+                config_bounds[i] < config_bounds[i + 1]
+                for i in range(len(config_bounds) - 1)
+            ):
+                bounds = config_bounds
+            else:
+                logger.warning(
+                    "Bounds must be all different and sorted in increasing"
+                    " order. Using default."
+                )
+
+        self.current = OrderedDict([(bb, 0) for bb in bounds])
+        self.current[inf] = 0
```

Comment:
Curious what does `inf` look like when printed?

Reply:
It looks like this:

```python
In [1]: from math import inf

In [2]: print(inf)
inf

In [3]: str(inf)
Out[3]: 'inf'
```